coral-service/src/main/java/com/linkedin/coral/coralservice/metastore/MetastoreProvider.java

```diff
@@ -0,0 +1,125 @@
+        CoralUtils.class.getClassLoader().getResourceAsStream("hive.properties");
+    final Properties props = new Properties();
+    props.load(hiveConfStream);
+    return new HiveMscAdapter(MetastoreProvider.getGridMetastoreClient(props));
+  }
+
+  /**
+   * Get metastore client for the hive metastore on Grid. Supports kerberos based
+   * authentication. Using kerberos requires user to get kerberos ticket and setup
+   * keytab externally.
+   * @param props hive configuration to use for connecting to the metastore
+   * @return metastore client
+   * @throws IOException indicates failure to read keytab file
+   * @throws MetaException indicates failure to instantiate metastore client
+   * @throws ConfigurationException on missing required configuration parameter
+   */
+  public static IMetaStoreClient getGridMetastoreClient(Properties props)
+      throws IOException, MetaException, ConfigurationException {
+
+    HiveConf conf = new HiveConf();
+    conf.setVar(METASTOREURIS, props.getProperty(HIVE_METASTORE_URIS, DEFAULT_METASTORE_URI));
+
+    if (props.getProperty(HIVE_METASTORE_AUTHENTICATION, DEFAULT_METASTORE_AUTHENTICATION)
+        .equalsIgnoreCase(KERBEROS_AUTHENTICATION)) {
+
+      String clientPrincipal = props.getProperty(HIVE_METASTORE_CLIENT_PRINCIPAL);
+      if (Strings.isNullOrEmpty(clientPrincipal)) {
+        throw new ConfigurationException(String.format("%s is required", HIVE_METASTORE_CLIENT_PRINCIPAL));
+      }
+      String clientKeytab = props.getProperty(HIVE_METASTORE_CLIENT_KEYTAB);
+      if (Strings.isNullOrEmpty(clientKeytab)) {
+        throw new ConfigurationException(String.format("%s is required", HIVE_METASTORE_CLIENT_KEYTAB));
+      }
+      String servicePrincipal = props.getProperty(HIVE_METASTORE_SERVICE_PRINCIPAL);
+      if (Strings.isNullOrEmpty(servicePrincipal)) {
+        throw new ConfigurationException(String.format("%s is required", HIVE_METASTORE_SERVICE_PRINCIPAL));
+      }
+      // leave this commented...handy debugging option rather than searching
+      // System.setProperty("sun.security.krb5.debug", "true");
+      conf.setVar(METASTORE_USE_THRIFT_SASL, "true");
+      conf.setVar(METASTORE_KERBEROS_PRINCIPAL, servicePrincipal);
+      conf.set(HADOOP_SECURITY_AUTHENTICATION, KERBEROS_AUTHENTICATION);
+
+      UserGroupInformation.setConfiguration(conf);
+      UserGroupInformation.loginUserFromKeytab(clientPrincipal, clientKeytab);
+    }
+    return new HiveMetaStoreClient(conf);
+  }
+
+  /**
+   * Get hive metastore client for local hive. Uses locally setup database
+   * and uses embedded derby to connect.
+   * @param props properties to connect to the local store
+   * @return Local hive metastore client
+   * @throws HiveException on failure to instantiate local hive
+   * @throws MetaException on failure to create local hive metastore
+   */
+  public static IMetaStoreClient getLocalMetastoreClient(Properties props) throws HiveException, MetaException {
```

Comment:
Similar feedback here, about using @beans.

Two objects here `GridMetastoreClient` and `LocalMetastoreClient`, one interface `IMetaStoreclient`. While booting the spring application, a decision can be made based on a configuration file (say coralconfigs.yaml) and appropriate object should be injected.